pkg/utils/utils.go

```diff
@@ -353,10 +366,20 @@ func GetS3PETMFromParamsMap(params map[string]interface{}, logger logrus.FieldLo
 		if !ok {
 			return nil, errors.New("Failed to retrieve S3 Secret Access Key.")
 		}
-		logger.Infof("Using explicitly found credentials for S3 repository access.")
+		credential = credentials.NewStaticCredentials(s3AccessKeyId, s3SecretAccessKey, "")
+		sessionOptions.Config.Credentials = credential
+	}
+	if len(caCert) > 0 {
```

Comment:
you don't need to initialize new session on different cases. Instead, you might want to accumulate different session options in different cases and initialize a new session based on different session options in one place. FYI, config is one of the field of session option.